code/modules/power/supermatter/supermatter.dm

```diff
@@ -630,6 +642,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			power_transmission_bonus += gas_comp[gasID] * gas_trans[gasID] * (isnull(transit_mod[gasID]) ? 1 : transit_mod[gasID])
 		power_transmission_bonus *= h2obonus
 
+		//Miasma is really just microscopic particulate. It gets consumed like anything else that touches the crystal.
+		if(gas_comp[/datum/gas/miasma])
+			var/miasma_pp = env.return_pressure() * gas_comp[/datum/gas/miasma]
+			var/consumed_miasma = clamp(((miasma_pp - MIASMA_CONSUMPTION_PP) / (miasma_pp + MIASMA_PRESSURE_SCALING)) + gasmix_power_ratio, MIASMA_CONSUMPTION_RATIO_MIN, MIASMA_CONSUMPTION_RATIO_MAX) * gas_comp[/datum/gas/miasma] * combined_gas
```

Comment:
Am I correct in my assessment of this function that it looks about like this?
```
var/consumed_miasma = clamp(ratio infinitely approaches 1 with higher molar counts. high pressure makes it approach 1 faster + gasmix_power_ratio, 0, 1)

consumed_miasma = (consumed_miasma) * miasma count * total count
```

Could you explain why this is doing what it's doing?

Reply:
The miasma composition is the ratio of miasma in the air by mole count. We multiply that by the total moles of gas to get the moles of miasma. The scaling factor is so that it consumes progressively more of the miasma in the air as the partial pressure of the miasma increases. It should approach consuming 100% of the miasma as the pressure goes up.

Reply:
The gasmix power ratio bit is in there just as a random bit of scaling that isn't dependent on miasma specifically. I honestly threw that in because I wanted to have miasma interact with the other gases and it happened to be a preexisting value in about the right scaling range.

Reply:
Yeah. You've got the right idea.

Reply:
I'm honestly not sure I like the gasmix_power_ratio, since by nature this behavior lowers the amount of miasma available to the sm, if there's a positive value like say o2 in the mix, the miasma percent consumed will be close to static, but if n2 is high it'll get lower. Kinda worried it'll make the gas come off as useless at first blush, thoughts?

Everything else looks good to me

Reply:
Yeah. You're right about the gasmix ratio. I could remove it, but then that would keep miasma from interacting with the other gases... Eh, I'll just get the base behavior in now and worry about nuance later.

Reply:
why not make miasma contribute a bit of the gasmix ratio, and use the ratio itself to scale miasma consumption slightly? maybe by a 3rd? that way it'll never do nothing, and more miasma is always at least slightly better then miasma + n2, but you can still build on it

Reply:
Alright. That sounds good.